estate/models/estate_property.py

```diff
@@ -0,0 +1,102 @@
+        ("check_expected_price","CHECK(expected_price >= 0)","Expected Price must be a positive amount")
+    ]
+    
+    #---------------------------------------------Basic Fields----------------------------------------#
+    name = fields.Char(required=True, default='MY HOME')
+    description = fields.Text(default="Big House in Mumbai") 
+    postcode = fields.Char(default='123456')
+    date_availability = fields.Date(default=lambda self:fields.Date.today() + relativedelta(months=3), copy=False)
+    expected_price = fields.Float(required=True)
+    selling_price = fields.Float(copy=False, readonly=True)
+    bedrooms = fields.Integer(default=2)
+    facades = fields.Integer(default=1)
+    is_garage = fields.Boolean(default=False)
+    is_garden = fields.Boolean(default=False)
+    active = fields.Boolean(default=True)
+
+    #--------------------------------------------Selection Fields---------------------------------------#
+    garden_orientation = fields.Selection(
+        selection=[('north', 'North'), ('south', 'South'), ('east', 'East'), ('west', 'West')],
+        string="Garden Orientation"
+    )   
+    state = fields.Selection([
+        ('new', 'New'), ('offer_received', 'Offer Received'), ('offer_accepted', 'Offer Accepted'),
+        ('sold', 'Sold'), ('cancelled', 'Cancelled')
+    ], default='new')
+    
+    measurement_unit = fields.Selection([
+        ('sqm', 'Square Meters'),
+        ('sqft', 'Square Feet')
+    ], default='sqm', string="Measurement Unit")  
+
+    #-------------------------------------------Computed Fields--------------------------------------------# 
+    living_area = fields.Float(string="Living Area", default = 0.0)
+    garden_area = fields.Float(string="Garden Area", default = 0.0)
+    total_area = fields.Float(compute = "_compute_total_area", string= "Total Area")
+    best_price = fields.Float(compute="_compute_best_price", string= "Best Price")
+
+    #--------------------------------------------Relational Fields------------------------------------------#
+    property_type_id = fields.Many2one("estate.property.type", string="Property Type")
+    buyer_id = fields.Many2one("res.partner", string="Buyer", readonly=True, copy=False)
+    seller_id = fields.Many2one("res.users", default=lambda self: self.env.user, string="Salesman")  
+    tag_ids = fields.Many2many("estate.property.tag", string="Tags")  
+    offer_ids = fields.One2many("estate.property.offer", "property_id", string="Offers")
+    
+    # --------------------------------------------Compute Methods----------------------------------------#
+    @api.depends("living_area", "garden_area", "measurement_unit")
+    def _compute_total_area(self):
+        for property in self:
+            property.total_area = property.living_area + property.garden_area
+            if property.measurement_unit == 'sqft':
+                property.total_area *= 10.764
+
```

Comment:
```suggestion
    @api.depends("living_area", "garden_area")
    def _compute_total_area(self):
        for property in self:
            property.total_area = property.living_area + property.garden_area
```
Good thinking and creativity.
But if i have a selection i will directly select that yeah i'm writing it in sqft or in sqm and enter the value in that measure

Reply:
yes sir , I understood your point and I'll implement the above suggested changes

Reply:
Complete this point